rules/otp.bzl

```diff
@@ -211,7 +100,14 @@ def _otp_image(ctx):
 otp_image = rule(
     implementation = _otp_image,
     attrs = {
-        "src": attr.label(allow_single_file = True),
+        "src": attr.label(
+            allow_single_file = [".json", ".hjson"],
+            doc = "Image configuration file in Hjson format.",
+        ),
+        "add_cfg": attr.label_list(
+            allow_files = [".json", ".hjson"],
+            doc = "Additional image configuration file(s) in Hjson format.",
+        ),
```

Comment:
WDYT about getting rid of `add_cfg` and making `src` an `attr.label_list`? Conceptually, this `otp_image` rule layers each JSON image config onto the previous one.

Something like this:

```bazel
        "image_configs": attr.label_list(
            allow_empty = False,
            allow_files = [".json", ".hjson"],
            doc = "Sequence of image configs in Hjson format. Later configs may clobber values from earlier configs.",
        ),
```

Reply:
Yeah I considered consolidating all these inputs as `srcs`, but `gen-otp-img.py` treats the "base" configuration file slightly differently from the overlay files (i.e. those passed in via `--add_cfg`). For example, it requires that the base configuration have a `seed` parameter[^1].

I thought it would be better to explicitly have a separate parameter for the base configuration rather than doing something like "the first element of the `srcs` attribute will be used as the base configuration".

[^1]: Creating the image with the base `img_cfg`: https://cs.opensource.google/opentitan/opentitan/+/master:util/design/lib/OtpMemImg.py;l=150
Applying the overlay from `add_cfg`:
https://cs.opensource.google/opentitan/opentitan/+/master:util/design/lib/OtpMemImg.py;l=325

Reply:
Since msf refers to the additional configs as layers, perhaps `layers` as the attribute name?

Reply:
Indeed - only the base config is allowed to have a seed value at the moment.
I believe this was done due to the way these additional config overlays are currently processed in the Python script.

I.e., the initial config is used to create the OTP image object and that involves seeding the RNG.
Additional overlays are then processed later on as "patches" to the base image - so changing the seed at a later point would interfere with how random constants are currently generated. We could loosen this constraint, but that would require a bit of refactoring so that the Python script first validates and combines all Hjson datastructures before creating an OTP image object (in hindsight this may have been a more efficient way of doing it).

Reply:
> Since msf refers to the additional configs as layers, perhaps layers as the attribute name?

Or maybe `overlays`?

Reply:
Changed the attr name to `overlays`